src/sys/mod.rs

```diff
@@ -1,3 +1,6 @@
+#[cfg(all(target_os = "linux",
+          any(target_arch = "x86", target_arch = "x86_64")))]
```

Comment:
can you add a comment that the gating is because libc doesn't include it for other platform / architectures? the functions themselves are part of POSIX: http://pubs.opengroup.org/onlinepubs/009695399/basedefs/ucontext.h.html